frontend/test/metabase/scenarios/question/filter.cy.spec.js

```diff
@@ -223,4 +225,113 @@ describe("scenarios > question > filter", () => {
+        },
+        database: 1,
+      },
+      display: "pie",
+      visualization_settings: {},
+    }).then(({ body: { id: QUESTION_ID } }) => {
+      cy.log("**--2. Create a dashboard--**");
+
+      cy.request("POST", "/api/dashboard", {
+        name: "13960D",
+      }).then(({ body: { id: DASHBOARD_ID } }) => {
+        cy.log(
+          "**--3. Add filters to the dashboard and set the default value to the first one--**",
+        );
+
+        cy.request("PUT", `/api/dashboard/${DASHBOARD_ID}`, {
+          name: "13960D",
+          parameters: [
+            {
+              name: "Category",
+              slug: "category",
+              id: "c32a49e1",
+              type: "category",
+              default: ["Doohickey"],
+            },
+            { name: "ID", slug: "id", id: "f2bf003c", type: "id" },
+          ],
+        });
+
+        cy.log("**--4. Add question to the dashboard--**");
+
+        cy.request("POST", `/api/dashboard/${DASHBOARD_ID}/cards`, {
+          cardId: QUESTION_ID,
+        }).then(({ body: { id: DASH_CARD_ID } }) => {
+          cy.log("**--5. Connect the filters to the card--**");
+
+          cy.request("PUT", `/api/dashboard/${DASHBOARD_ID}/cards`, {
+            cards: [
+              {
+                id: DASH_CARD_ID,
+                card_id: QUESTION_ID,
+                row: 0,
+                col: 0,
+                sizeX: 10,
+                sizeY: 10,
+                series: [],
+                visualization_settings: {},
+                parameter_mappings: [
+                  {
+                    parameter_id: "c32a49e1",
+                    card_id: QUESTION_ID,
+                    target: ["dimension", ["field-id", PRODUCTS.CATEGORY]],
+                  },
+                  {
+                    parameter_id: "f2bf003c",
+                    card_id: QUESTION_ID,
+                    target: ["dimension", ["field-id", PRODUCTS.ID]],
+                  },
+                ],
+              },
+            ],
+          });
+        });
+        cy.server();
+        cy.route("POST", `/api/card/${QUESTION_ID}/query`).as("cardQuery");
+
+        cy.visit(`/dashboard/${DASHBOARD_ID}`);
+
+        cy.wait("@cardQuery");
+        cy.location("search").should("eq", "?category=Doohickey");
+
+        // Remove default filter (category)
+        cy.get("fieldset .Icon-close").click();
+
+        cy.url().should("not.include", "?category=Doohickey");
+
+        // Set filter value to the `ID`
+        cy.get("fieldset")
+          .contains(/ID/i)
+          .click();
+        cy.findByPlaceholderText("Enter an ID").type("1");
+        cy.findByText("Add filter")
+          .closest("button")
+          .should("not.be.disabled")
+          .click();
+
+        cy.location("search").should("eq", "?id=1");
+
+        // Reload" page, or rather visit the dashboard link again
+        cy.visit(`/dashboard/${DASHBOARD_ID}`);
```

Comment:
That's not a reload. This will always go to `/dashboard/${DASHBOARD_ID}?category=Doohickey` and never include the `id=1`.
```suggestion
        cy.visit(`/dashboard/${DASHBOARD_ID}?id=1`);
```
And I don't know if you are going to hit a race-condition with the assertion, since default filter values are applied a split-second after the dashboard is already loaded.

Reply:
I was avoiding `cy.reload()` because it always results in XHR being aborted. But you're absolutely right - it was silly to go to the dashboard itself - that didn't replicate anything.

I'm changing it back to `cy.reload()`.

Reply:
> I'm unsure what the fix for the issue would actually be. Perhaps it would leave the default parameter in the URL without a value.

For starters, I'd expect NOT to see "Doohickey" value set to the first filter. Leaving that value in the url also doesn't make sense to me.